### DISCUSSION TITLE
Taiko return code

### DISCUSSION THREAD

**Discussion by jbjulien** [getgauge/taiko#1766] [General]
Hello,
To integrate with other test tools, is it possible for taiko to set a return code <> 0 when there is an error ? I tried to add a return 1 in the catch error block, but taiko seems to always finish with return code 0.
Thank you
Jean-Baptiste

**Comment by NivedhaSenthil** [comment]
can you please elaborate on your scenario ? If to be used with other test runner, Taiko can be used just as a simple module and when a error occurs its the runners responsibility to handle it.

**Comment by jbjulien** [comment]
Thank you Nivedha, I am new to js scripting.
My scenario is to run taiko scripts inside the chaos toolkit tool (via process provider that launch a local process on the host). choas toolkit knows if the step is OK or KO with the return code.
From your answer I understand that I could write a js script wrapper that would call taiko as a module.
Thanks
Jean-Baptiste

**Reply by NivedhaSenthil** [reply]
yup taiko can be used with any other test runner like Gauge, mocha, jest etc., also taiko's bundled minimal runner exits with a non-zero exit code unless the error is caught and handled by the user

**Reply by jbjulien** [reply]
Thank you Nivedha, I was catching the error without using the proper exit statement : works well with process.exitCode = 1; 
My lack of knowledge of js scripting !
